### PR TITLE
Added `HierarchicalNameMapper` as a dependency of `StatsdMeterRegistry`

### PR DESCRIPTION
**Pull request by dheld-expedia**
* Defaults to `HierarchicalNameMapper.DEFAULT` if none is provided

### CODE REVIEW

File: implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java [hunk 1]
```diff
@@ -15,8 +15,19 @@
  */
 package io.micrometer.statsd;
 
-import io.micrometer.core.instrument.*;
+import io.micrometer.core.instrument.Clock;
```
Comment (dheld-expedia) [inline implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java]: My IntelliJ settings do this automatically, but if you prefer the wildcards, I will undo this.

**Comment by dheld-expedia** [comment]
I believe this is the minimum change possible to allow custom configuration of a HierarchicalNameMapper for the StatsD back-end.

File: implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java (hunk 1 above)
Comment (spencergibb) [reply implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java]: We generally prefer non `*` imports.

File: implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdLineBuilder.java [hunk 2]
```diff
@@ -61,9 +60,10 @@
         return numberFormatter;
     });
 
-    StatsdLineBuilder(Meter.Id id, StatsdFlavor flavor, NamingConvention convention) {
+    StatsdLineBuilder(Meter.Id id, StatsdFlavor flavor, HierarchicalNameMapper nameMapper, NamingConvention convention) {
```
Comment (spencergibb) [inline implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdLineBuilder.java]: Maybe overload the constructor with `HierarchicalNameMapper.DEFAULT` as the default.

Comment (dheld-expedia) [reply implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdLineBuilder.java]: Yeah...I thought that `build` was the default task for the project, so I didn't actually run the tests before pushing. :/

File: implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java (hunk 1 above)
Comment (jkschneider) [reply implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java]: Ah, not in Micrometer @spencergibb. It's the IntelliJ default, and they are everywhere ;)

**Comment by dheld-expedia** [comment]
I believe this implements https://github.com/micrometer-metrics/micrometer/issues/156, but I'm not sure what the intended solution there was.

File: build.gradle [hunk 3]
```diff
@@ -48,3 +48,5 @@ subprojects {
 task wrapper(type: Wrapper) {
     gradleVersion = '4.0.1'
 }
+
+defaultTasks 'build'
```
Comment (checketts) [inline build.gradle]: Was this change intentionally checked in?

Comment (dheld-expedia) [reply build.gradle]: Yup.  It will make Gradle do the thing which most devs will probably expect.  Do you think it's bad?

Comment (checketts) [reply build.gradle]: I wasn't familiar with that feature.

I think `'clean', 'build'` might be a better default

**Comment by dheld-expedia** [comment]
That is true if the value is in [0.0, 1.0].  However, the value I get is -1.0, because the `java.lang.management` version of the bean does not implement this call on Windows.

http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6336608

To prove it, I ran this little test, which printed -1.0 for as long as I was willing to wait:
```
        final OperatingSystemMXBean operatingSystemBean = ManagementFactory.getOperatingSystemMXBean();
        while (operatingSystemBean.getSystemLoadAverage() <= 0) {
            System.out.println("Waiting for load average: " + operatingSystemBean.getSystemLoadAverage());
            sleep(1000);
        }
```

**Comment by dheld-expedia** [comment]
If you are willing to provide a Sun JVM-specific solution for Windows, there is this option: https://docs.oracle.com/javase/7/docs/jre/api/management/extension/com/sun/management/OperatingSystemMXBean.html#getSystemCpuLoad().  I guess you could try to load the class and only use it if it's present.

File: build.gradle (hunk 3 above)
Comment (dheld-expedia) [reply build.gradle]: Well, in a regular dev cycle, you will want to call `build` many times, but `clean` only rarely.  I'd rather not pay the cost of `clean` every time, and having to specify `build` every time defeats the purpose of making it the default!

Comment (checketts) [reply build.gradle]: Sounds good

**Comment by jkschneider** [comment]
@dheld-expedia Wow, thanks for the research. I had no idea.

**Pull request merged**